godtools/App/Share/Data/ArticlesRepository/Repository/Downloader/ArticleAemDownloadGetCacheTimeInterval.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  ArticleAemDownloadGetCacheTimeInterval.swift
+//  godtools
+//
+//  Created by Levi Eggert on 3/28/23.
+//  Copyright © 2023 Cru. All rights reserved.
+//
+
+import Foundation
+
+class ArticleAemDownloadGetCacheTimeInterval {
+    
+    init() {
+        
+    }
+    
+    func getCacheTimeInterval(cachePolicy: ArticleAemDownloadOperationCachePolicy) -> TimeInterval? {
+        
+        let calendar: Calendar = Calendar.current
+        let currentDate: Date = Date()
+        let currentDateComponents: DateComponents = calendar.dateComponents([.month, .day, .year, .hour, .minute, .second], from: currentDate)
+        
+        switch cachePolicy {
+            
+        case .fetchFromCacheUpToNextHour:
+            
+            guard let currentMinutes = currentDateComponents.minute, let currentSeconds = currentDateComponents.second else {
+                return nil
+            }
+            
+            let minutesUpToNextHour: Int = 60 - currentMinutes
+            
+            var dateUpToNextHourComponents: DateComponents = DateComponents()
+            dateUpToNextHourComponents.minute = minutesUpToNextHour
+            dateUpToNextHourComponents.second = currentSeconds * -1
+            
+            guard let dateUpToNextHour = calendar.date(byAdding: dateUpToNextHourComponents, to: currentDate) else {
+                return nil
+            }
+            
+            let timeInterval: TimeInterval = dateUpToNextHour.timeIntervalSince1970
+            
+            return timeInterval
```

Comment:
no way to get seconds/milliseconds since epoch? if you have that you can just do a fairly simple `secs - (secs % secs_in_hour)` or an equivalent for milliseconds.

Reply:
how I do the actual rounding on Android
https://github.com/CruGlobal/godtools-android/blob/develop/ui/article-aem-renderer/src/main/kotlin/org/cru/godtools/article/aem/service/AemArticleManager.kt#L346-L348

Reply:
Hmm, I didn't think of that.  That would be less code.  

Seconds since epoch would be ```Date().timeIntervalSince1970```.

https://developer.apple.com/documentation/foundation/nsdate/1407504-timeintervalsince1970

Reply:
Going to update.

Reply:
```swift
var timestamp = Date().timeIntervalSince1970
if (!force) {
    timestamp = timestamp - (timestamp % 3600)
}
return timestamp
```